js/src/admin/components/ExtensionPage.js

```diff
@@ -75,10 +77,12 @@ export default class ExtensionPage extends Page {
           </div>
           <div className="helpText">{this.extension.description}</div>
           <div className="ExtensionPage-headerItems">
-            <Switch state={this.isEnabled()} onchange={this.toggle.bind(this, this.extension.id)}>
-              {this.isEnabled(this.extension.id)
-                ? app.translator.trans('core.admin.extension.enabled')
-                : app.translator.trans('core.admin.extension.disabled')}
+            <Switch
+              state={this.changingState ? !isEnabled : isEnabled}
+              loading={this.changingState}
+              onchange={this.toggle.bind(this, this.extension.id)}
+            >
+              {isEnabled ? app.translator.trans('core.admin.extension.enabled') : app.translator.trans('core.admin.extension.disabled')}
```

Comment:
Shouldn't the switch and the label display the same value?

Reply:
I don't think so. The text shows whether the extension is currently enabled or disabled, while the Switch can show that the extension is being enabled/disabled through it's loading spinner. The switch is visible behind the modal, while the text not so much. 

I don't think the text should change at all.

Reply:
Ah got it, I see your point.